diff -Naur boost_1_45_0.old/tools/build/v2/tools/darwin.jam boost_1_45_0/tools/build/v2/tools/darwin.jam
--- boost_1_45_0.old/tools/build/v2/tools/darwin.jam	2013-02-19 16:31:36.000000000 -0800
+++ boost_1_45_0/tools/build/v2/tools/darwin.jam	2013-02-19 16:32:38.000000000 -0800
@@ -482,7 +482,7 @@
 flags darwin.compile OPTIONS <link>shared : -dynamic ;
 
 # Misc options.
-flags darwin.compile OPTIONS : -no-cpp-precomp -gdwarf-2 ;
+flags darwin.compile OPTIONS : -gdwarf-2 ;
 
 # Add the framework names to use.
 flags darwin.link FRAMEWORK <framework> ;
